pkg/trace/writer/stats.go

```diff
@@ -36,22 +36,26 @@ const (
 
 // StatsWriter ingests stats buckets and flushes them to the API.
 type StatsWriter struct {
-	in        <-chan []stats.Bucket
-	inPayload chan *stats.Payload
-	hostname  string
-	env       string
-	senders   []*sender
-	stop      chan struct{}
-	stats     *info.StatsWriterInfo
+	// InBuckets is the input channel for stats buckets.
+	InBuckets chan []stats.Bucket
+
+	// InPayload is the input channel for payloads.
+	InPayload chan *stats.Payload
+
+	hostname string
+	env      string
+	senders  []*sender
+	stop     chan struct{}
+	stats    *info.StatsWriterInfo
 
 	easylog *logutil.ThrottledLogger
 }
 
 // NewStatsWriter returns a new StatsWriter. It must be started using Run.
-func NewStatsWriter(cfg *config.AgentConfig, inBuckets <-chan []stats.Bucket, inPayloads chan *stats.Payload) *StatsWriter {
+func NewStatsWriter(cfg *config.AgentConfig, inBuckets chan []stats.Bucket) *StatsWriter {
 	sw := &StatsWriter{
-		in:        inBuckets,
-		inPayload: inPayloads,
+		InBuckets: inBuckets,
```

Comment:
For consistency shouldn't inBuckets be created here too?

Reply:
The problem with it is that it's shared with the concentrator here: https://github.com/DataDog/datadog-agent/blob/a4ccdcf1d97fe803e421f803a6646092a2a2e5a9/pkg/trace/agent/agent.go#L73-L82